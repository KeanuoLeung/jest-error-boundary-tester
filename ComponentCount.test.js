import renderer from 'react-test-renderer';
import Link from './src/App';

const TARGET = 'ErrorBoundary';

it('changes the class when hovered', () => {
  const component = renderer.create(
    <Link page="http://www.facebook.com">Facebook</Link>
  );
  let tree = component.getInstance();
  const componentNames = [];
  const test = { ...component.root._fiber };
  test._parent = component.root._fiber;
  const nodeList = [test];
  let curNode;
  function findNearestParentName(node) {
    let curNode = node;
    while (!curNode?.type?.name) {
      curNode = curNode._parent;
    }
    return curNode?.type?.name;
  }
  while ((curNode = nodeList.pop())) {
    if (curNode.child) {
      nodeList.push({ ...curNode.child, _parent: curNode });
    }
    if (curNode.sibling) {
      nodeList.push({ ...curNode.sibling, _parent: curNode._parent });
    }

    if (curNode?.type?.name === TARGET) {
      componentNames.push({
        name: curNode?.type?.name ?? curNode?.type,
        parent: findNearestParentName(curNode?._parent),
      });
    }
  }

  console.log('component names', componentNames);
  console.log('this is tree!!!', component.root._fiber.child);
});
