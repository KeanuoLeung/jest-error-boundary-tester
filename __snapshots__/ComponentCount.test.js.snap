// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the class when hovered 1`] = `
<div
  className="App"
>
  <h1>
    Hello CodeSandbox
  </h1>
  <div>
    Error
  </div>
  <h2>
    Start editing to see some magic happen!
  </h2>
</div>
`;
